### PR TITLE
add update endorser list function.

### PR DESCRIPTION
**Pull request by yenkhoon**
- added update endorser list function
- added error message for updating endorser list
- added error message for token item id is in used
- added missing field memo in update token metadata

### CODE REVIEW

**Comment by yenkhoon** [comment]
@jeansoonsik help review, i have added update nft endorser list.

**Comment by yenkhoon** [comment]
@jeansoonsik added test cases and error code for update endorser list and update item metadata.

File: src.ts/non-fungible-token.ts [hunk 1]
```diff
@@ -443,14 +443,64 @@ export class NonFungibleToken {
...
+                if (overrides && overrides.sendOnly) {
+                    return response;
+                }
+                let confirmations = (overrides && overrides.confirmations) ? Number(overrides.confirmations) : null;
+
+                return this.signer.provider.waitForTransaction(response.hash, confirmations).then((receipt) => {
+                    if (1 == receipt.status) {
+                        return receipt;
+                    }
+                    throw this.signer.provider.checkTransactionReceipt(receipt, errors.CALL_EXCEPTION, "update non fungible token item metadata failed", {
```
Comment (jeansoonsik) [inline src.ts/non-fungible-token.ts:472]: incorrect error message

File: src.ts/non-fungible-token.ts [hunk 2]
```diff
@@ -443,14 +443,64 @@ export class NonFungibleToken {
...
+                        receipt
+                    });
+                });
+            });
+        });
+    }
+
+    getUpdateEndorserListTransactionRequest(endorsers: string[], overrides?: any): Promise<TransactionRequest> {
+        if (!this.signer) {
+            errors.throwError('update non fungible token item metadata require signer', errors.NOT_INITIALIZED, { arg: 'signer' });
```
Comment (jeansoonsik) [inline src.ts/non-fungible-token.ts:483]: incorrect error message

File: src.ts/providers/json-rpc-provider.ts [hunk 3]
```diff
@@ -535,6 +535,10 @@ function checkResponseLog(self: JsonRpcProvider, method: string, result: any, de
                         return errors.createError('invalid token owner', errors.NOT_ALLOWED, { operation: method, info, response: result, params });
                     case 2104: // Transfer token ownership approved
                         return errors.createError('Token ownership is already approved', errors.EXISTS, { operation: method, info, response: result, params });
+                    case 2105:
+                        return errors.createError('Token item id is in used', errors.EXISTS, { operation: method, info, response: result, params });
+                    case 2106:
+                        return errors.createError('Token item endorser invalid', errors.EXISTS, { operation: method, info, response: result, params });
```
Comment (jeansoonsik) [inline src.ts/providers/json-rpc-provider.ts:541]: the error code should be errors.NOT_ALLOWED

Comment (yenkhoon) [reply src.ts/providers/json-rpc-provider.ts]: fixed

**Pull request merged**